### PR TITLE
Client interceptor: link parent span from SpanManager

### PR DESCRIPTION
**Pull request by pavolloffay**
@objectiser @jpkrohling could you please review?

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay LGTM - although possibly the itest changes should have been a separate PR.

File: opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptor.java [hunk 1]
```diff
@@ -53,10 +56,11 @@ public ClientHttpResponse intercept(HttpRequest httpRequest, byte[] body,
         Tracer.SpanBuilder spanBuilder = tracer.buildSpan(httpRequest.getMethod().toString())
                 .withTag(Tags.SPAN_KIND.getKey(), Tags.SPAN_KIND_CLIENT);
 
-        // TODO add parent span via in-process propagation
-//        if (parent-is-present) {
-//            spanBuilder.asChildOf(parent);
-//        }
+        // link with parent span
+        SpanManager.ManagedSpan parentSpan = spanManager.current();
```
Comment (jpkrohling) [inline opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptor.java]: `current()` is guaranteed to not be null, right?

Comment (objectiser) [inline opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptor.java]: That is correct.

Comment (pavolloffay) [inline opentracing-spring-web-handler-interceptor/src/main/java/io/opentracing/contrib/spring/web/client/TracingRestTemplateInterceptor.java]: yes not null, but wrapped span can be null.

**Comment by pavolloffay** [comment]
@objectiser @jpkrohling thanks for review!

> @pavolloffay LGTM - although possibly the itest changes should have been a separate PR.

agree, I will keep it here, tests are not deployed to maven so it shouldn't hurt anybody.

**Pull request merged**